### ISSUE TITLE
radar did not find any object!

### ISSUE THREAD

**Issue by Amin-Golden** [frk2/toyoyta_radar_control_can#2]
Hi
I bought a stock Toyota 8821007010 radar from eBay and used your code to emulate a Toyota car to read radar data, But the radar did not find any object. What do you think I should do?

**Comment by SamSkjord** [comment]
I also ended up with the same unit, my research seems to show it's from later vehicles that use TSS-P and it's looking for signals from the DSU to activate.

Believe we need the 88210-47090, have one on order and will update after testing

**Comment by SamSkjord** [comment]
> Believe we need the 88210-47090, have one on order and will update after testing

no change, just _[RX] ID: 0x4FF Data: 3F00000200000000 (8 bytes)_
both of mine have been UK/RHD units and I beleive frk2's is LHD

**Comment by SamSkjord** [comment]
...cancel last s
swap the canbusses and both of mine fire right up, I'll be publishing my repo shortly

**Comment by Amin-Golden** [comment]
> ...cancel last s swap the canbusses and both of mine fire right up, I'll be publishing my repo shortly

Great news can you guide me to solve the problem ?

**Comment by SamSkjord** [comment]
Of course; what's your setup? how are you sending can messages etc?

**Comment by Amin-Golden** [comment]
I have a stock Toyota 8821007010 radar and I connect both CAN to a comma ai white panda and use this repo to read radar data, I receive all radar info but all objects info are zeros.

**Comment by SamSkjord** [comment]
Ah, I don't have any comma/openpilot hardware.
Are you using https://github.com/wocsor/Toyota-Radar-Spoofer?

**Comment by Amin-Golden** [comment]
> Ah, I don't have any comma/openpilot hardware. Are you using https://github.com/wocsor/Toyota-Radar-Spoofer?

Yes I tried that on too. Can you explain your setup and what codes do you use?

**Comment by SamSkjord** [comment]
I'm using a Raspberry Pi 4 with a [Waveshare Dual CAN](https://www.waveshare.com/2-ch-can-hat-plus.htm) board 
frk2's code worked basically out of the box... once I got the busses the right way round

I've then had Codex write me a driver https://github.com/SamSkjord/toyota-radar-driver
it sends the initial messages and then the keepalive messages at 100hz on the car bus then the radar bus returns track data (when it has some, otherwise 0s for all tracks)
